### DISCUSSION TITLE
first: Header does not appear to be functioning

### DISCUSSION THREAD

**Discussion by devinrobbins** [dolanmiu/docx#1047] [Q&A]
I've been following the guides and documentation for this, but this functionality for the "first" header does not appear to be working, or maybe I'm misunderstanding.  Is ""first" meant to appear on the first page of the document or is there something else I'm missing?  Here's a code snippet to show how I have the Header section setup:     

sections: [
            {
                headers: {

                    default: new Header({     
                        children: [
                            new Paragraph({
                                children: [
                                    new TextRun({
                                        text: "Default header Text",
                                    }),
                                ],
                            }),
                        ],
                    }),

                    first: new Header({
                        children: [
                            new Paragraph({
                                children: [
                                    new TextRun({
                                        text: "first page header text",
                                    }),
                                ],
                            }),
                        ],
                    }),
                },

The default header will populate and appear throughout the document, regardless of the number of pages.  I'm using version 6.0.3. 

Thanks

**Comment by tadmi** [comment]
To reflect the title page in the section options add **titlePage: true**
```
const doc = new Document({
	sections: [
		{
			titlePage: true,
			children: [
				new Paragraph({
					children: [new TextRun('Hello World')]
				})
			]
		}
	]
});
```

**Reply by dishmael** [reply]
Thanks @tadmi, I got it working by creating two sections. Is that the correct way?

```
const doc = new Document({
  sections: [
    {
      titlePage: true,
      footers: {
        default: new Footer({
            children: [
              new Paragraph({
                text: "Title Page Footer",
              }),
            ],
          }),
      },
      children: [
        new Paragraph({
          children: [new TextRun('Title Page')]
        })
      ]
    },
    {
      footers: {
        default: new Footer({
            children: [
              new Paragraph({
                text: "Default Page Footer",
              }),
            ],
          }),
      },
      children: [
        new Paragraph({
          children: [new TextRun('First Page')]
        })
      ]
    },
  ]
});
```

**Comment by tadmi** [comment]
### example with 1 section on node js

```
const fs = require('fs');
const docx = require('docx');

const { Document, Packer, Paragraph, Footer, TextRun, PageBreak } = docx;

const doc = new Document({
	evenAndOddHeaderAndFooters: true,
	sections: [
		{
			properties: {
				titlePage: true
			},
			footers: {
				first: new Footer({
					children: [
						new Paragraph({
							text: 'footer on 1 page in the section'
						})
					]
				}),
				default: new Footer({
					children: [
						new Paragraph({
							text: 'footer on all other pages in the section or on odd pages if evenAndOddHeaderAndFooters = true'
						})
					]
				}),
				even: new Footer({
					children: [
						new Paragraph({
							text: 'footer on even pages in the section'
						})
					]
				})
			},
			children: [
				new Paragraph({
					children: [
						new TextRun('text on page 1'),
						new PageBreak(),
						new TextRun('text on page 2'),
						new PageBreak(),
						new TextRun('text on page 3')
					]
				})
			]
		}
	]
});

Packer.toBuffer(doc).then((buffer) => {
	fs.writeFileSync('doc.docx', buffer);
});
```

**Reply by dishmael** [reply]
That worked, thanks @tadmi!
